report/thesis.tex

```diff
@@ -977,12 +1033,20 @@ \chapter{Related problems and future work}
 \chapter{Conclusion}
 %%%%%%%%%%%%%%%%%%%%
 
-TODO
+In this work we have tackled the fork accountability problem from different points of view in order to better show all the theoretical and practical implications this challenge offers. From a high-level perspective, we have concretely defined the problem, analyzed all the scenarios where the problem can occur and analyzed the main misbehaviors a process can do to create the conditions of a fork.
+
+After tackling the problem, we described the design and the implementation of a fork accountability algorithm in Tendermint. We showed how it is possible to implement an algorithm when the communication between the monitor and validators is completely asynchronous, made the different algorithmic parts more concrete and took steps into implementing and getting a sense of the validity of this algorithm. We carefully studied and tested the proposed solution both from a correctness and a performance point of view, thanks to the numerous experiments and benchmarks that have been carried out, and proved that the algorithm is correct and efficient in terms of computational complexity. 
```

Comment:
Maybe mention again that you implemented the algorithm in Go.

Reply:
Fixed